test/e2e/shell.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var (
+	root = absOrDie(filepath.Clean(filepath.Join(path.Base(os.Args[0]), "..")))
+)
+
+var _ = Describe("Shell", func() {
+	// Slurp up all the tests in hack/e2e-suite
+	bashE2ERoot := filepath.Join(root, "hack/e2e-suite")
+	files, err := ioutil.ReadDir(bashE2ERoot)
+	if err != nil {
+		Fail(err.Error())
+	}
+
+	for _, file := range files {
+		It(fmt.Sprintf("test that %v passes", file.Name()), func() {
+			runBashTest(filepath.Join(bashE2ERoot, file.Name()))
+		})
+	}
+})
+
+func absOrDie(path string) string {
+	out, err := filepath.Abs(path)
+	if err != nil {
+		panic(err)
+	}
+	return out
+}
+
+// Runs the given bash test.
+func runBashTest(bash string) {
+	cmd := exec.Command("bash", "-s")
+	cmd.Stdin = strings.NewReader(bashWrapTest(bash))
+
+	stdout, stderr := bytes.NewBuffer(nil), bytes.NewBuffer(nil)
+	cmd.Stdout = stdout
+	cmd.Stderr = stderr
+
+	signals := make(chan os.Signal, 100)
+	done := make(chan struct{})
+	defer close(done)
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			case s := <-signals:
+				cmd.Process.Signal(s)
+			}
+		}
+	}()
+
+	if err := cmd.Run(); err != nil {
+		Fail(fmt.Sprintf("Error running %v:\nstdout:\n%v\nstderr:\n%v\n", bash, stdout, stderr))
+		return
+	}
+	return
+}
+
+func bashWrapTest(cmd string) string {
```

Comment:
Noooooo!!!

Seriously though, I'm working on another PR to make the hack/e2e-suite/*.sh stand on their own so this snippet will not be required then. I'd suggest holding this one until that one is in.